### PR TITLE
tests:kernel: added tests for v1.8

### PR DESCRIPTION
**Pull request by sharronliu**
Added kernel test cases for v1.8

Jira: ZEP-948
Jira: ZEP-1631

Signed-off-by: Sharron LIU <sharron.liu@intel.com>

### CODE REVIEW

File: tests/kernel/threads_scheduling/schedule_api/src/test_sched_timeslice_reset.c [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	/*disable timeslice*/
+	k_sched_time_slice_set(0, K_PRIO_PREEMPT(0));
+
+	for (int j = 0; j < 2; j++) {
+		k_sem_reset(&sema);
+		/* update priority for current thread*/
+		k_thread_priority_set(k_current_get(), K_PRIO_PREEMPT(j));
+		/* create delayed threads with equal preemptive priority*/
+		for (int i = 0; i < NUM_THREAD; i++) {
+			tid[i] = k_thread_spawn(tstack[i], STACK_SIZE,
```
Comment (nashif) [inline tests/kernel/threads_scheduling/schedule_api/src/test_sched_timeslice_reset.c]: please use k_thread_create instead

File: tests/kernel/threads_scheduling/schedule_api/src/test_sched_timeslice_reset.c [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	SLICE_SIZE      /* the spawned thread taking a full timeslice, reset*/
+};
+static int thread_idx;
+
+static void thread_tslice(void *p1, void *p2, void *p3)
+{
+	s64_t t = k_uptime_delta(&elapsed_slice);
+
+	TC_PRINT("thread[%d] elapsed slice %lld, ", thread_idx, t);
+	TC_PRINT("expected %lld\n", expected_slice[thread_idx]);
```
Comment (nashif) [inline tests/kernel/threads_scheduling/schedule_api/src/test_sched_timeslice_reset.c]: print things to the screen only in debug mode, the test output should  be clean without noise coming out of the test

**Review by nashif — CHANGES_REQUESTED** [review]

File: tests/kernel/threads_scheduling/schedule_api/src/test_sched_timeslice_reset.c (hunk 2 above)
Comment (sharronliu) [inline tests/kernel/threads_scheduling/schedule_api/src/test_sched_timeslice_reset.c]: done

File: tests/kernel/threads_scheduling/schedule_api/src/test_sched_timeslice_reset.c (hunk 1 above)
Comment (sharronliu) [inline tests/kernel/threads_scheduling/schedule_api/src/test_sched_timeslice_reset.c]: done

**Comment by nashif** [comment]
@sharronliu you probably want to squash the commit about timeslice into 1 commit and force push to your branch.

**Comment by sharronliu** [comment]
@Anas, I re-ordered these two commits (git rebase -i) and get the earlier one updated. Thanks.

**Review by nashif — APPROVED** [review]

**Pull request merged**